draft-ietf-quic-transport.md

```diff
@@ -1834,10 +1834,9 @@ SHOULD include information that allows the server to verify that the source IP
 address and port in client packets remains constant.
 
 Servers might use tokens from NEW_TOKEN in deciding not to send a Retry packet,
-even if the client address has changed.  A token that was provided in
-NEW_TOKEN cannot be used for address validation if the client address is not the
-same, though servers MAY allow for the possibility of changes arising from new
-mappings at a NAT.
+even if the client address has changed. Tokens sent in NEW_TOKEN frames SHOULD
+include information that allows the server to verify if the client address is
+stable, but might allow for different NAT bindings or ephemeral port selection.
```

Comment:
I don't really understand the second sentence. What's an example of information that would allow the server to verify if the client address is stable.

Reply:
== on IP and port would be one example, but the emphemeral port example clearly leads to just IP == IP.

Should we just say "IP MUST be the same" instead?  That's not good always (see CGNAT) and I want to avoid creating the impression that this is sufficient, even if it might be sufficient.

Reply:
Yeah one way of moving forward would be to state the intent along with how it might be implemented, rather than using RFC 2119 words for explaining how it should be implemented.

For example: "MUST validate that the network path between the peers have not changed (e.g., client's IP address is stable)."